packages/sdk/src/conversation/sso/ssoExecutionDialog.ts

```diff
@@ -0,0 +1,307 @@
+    ]);
+
+    this.commandMapping.set(commandId, triggerPatterns);
+    this.addDialog(dialog);
+  }
+
+  /**
+   * The run method handles the incoming activity (in the form of a DialogContext) and passes it through the dialog system.
+   *
+   * @param context The context object for the current turn.
+   * @param accessor The instance of StatePropertyAccessor for dialog system.
+   */
+  public async run(context: TurnContext, accessor: StatePropertyAccessor) {
+    const dialogSet = new DialogSet(accessor);
+    dialogSet.add(this);
+
+    const dialogContext = await dialogSet.createContext(context);
+    this.ensureMsTeamsChannel(dialogContext);
+    const results = await dialogContext.continueDialog();
+    if (results && results.status === DialogTurnStatus.empty) {
+      await dialogContext.beginDialog(this.id);
+    }
+  }
+
+  private getActivityText(activity: Activity): string {
+    let text = activity.text;
+    const removedMentionText = TurnContext.removeRecipientMention(activity);
+    if (removedMentionText) {
+      text = removedMentionText
+        .toLowerCase()
+        .replace(/\n|\r\n/g, "")
+        .trim();
+    }
+
+    return text;
+  }
+
+  private async commandRouteStep(stepContext: any) {
+    const turnContext = stepContext.context as TurnContext;
+
+    const text = this.getActivityText(turnContext.activity);
+
+    const commandId = this.matchCommands(text);
+    if (commandId) {
+      return await stepContext.beginDialog(commandId);
+    }
+    await stepContext.context.sendActivity(`Cannot find command: ${turnContext.activity.text}`);
+    return await stepContext.endDialog();
+  }
+
+  private async ssoStep(stepContext: any) {
+    try {
+      const turnContext = stepContext.context as TurnContext;
+
+      const text = this.getActivityText(turnContext.activity);
+      const message: CommandMessage = {
+        text,
+      };
+
+      stepContext.options.commandMessage = message;
+
+      return await stepContext.beginDialog(TEAMS_SSO_PROMPT_ID);
+    } catch (error: unknown) {
+      const context = stepContext.context;
+      await context.sendActivity("Failed to run SSO step");
+      await context.sendActivity((error as Error).message);
+      return await stepContext.endDialog();
+    }
+  }
+
+  private async dedupStep(stepContext: any) {
+    try {
+      const tokenResponse = stepContext.result;
+      // Only dedup after ssoStep to make sure that all Teams client would receive the login request
+      if (tokenResponse && (await this.shouldDedup(stepContext.context))) {
+        return Dialog.EndOfTurn;
+      }
+      return await stepContext.next({
+        tokenResponse,
+        message: stepContext.options.commandMessage,
+      });
+    } catch (error: unknown) {
+      const context = stepContext.context;
+      await context.sendActivity("Failed to run dedup step");
+      await context.sendActivity((error as Error).message);
+      return await stepContext.endDialog();
+    }
+  }
+
```

Comment:
we may save the raw text (without removing mention and other things), in case developer needs related information

Reply:
As discussed, we will keep it same logic with none sso command bot